fwdpy11/headers/fwdpy11/discrete_demography/DiscreteDemography.hpp

```diff
@@ -238,6 +238,73 @@ namespace fwdpy11
                     }
             }
 
+            void
+            validate_change_migration_events()
+            {
+                if (migmatrix == nullptr)
+                    {
+                        if (!set_migration_rates.empty())
+                            {
+                                throw std::invalid_argument(
+                                    "migration matrix is None but "
+                                    "SetMigrationRates events are registered");
+                            }
+                        return;
+                    }
+                for (auto& event : set_migration_rates)
+                    {
+                        if (event.migrates.size() == migmatrix->npops)
+                            {
+                                if (migmatrix->scaled == true
```

Comment:
This may not be a great change?  We may want to allow a zeroed-out change, for example if the source deme size is also set to zero that generation and the user is being cautious about other parameters.  It may be best to catch this error later, when building the migration lookup, and throw the MigrationError.